src/controller/CHIPDeviceController.cpp

```diff
@@ -1997,7 +2055,8 @@ void DeviceCommissioner::PerformCommissioningStep(DeviceProxy * proxy, Commissio
         app::InteractionModelEngine * engine = app::InteractionModelEngine::GetInstance();
         app::ReadPrepareParams readParams(proxy->GetSecureSession().Value());
 
-        app::AttributePathParams readPaths[8];
+        // NOTE: this array cannot have more than 9 entries, since 9 is what the spec requires as a minimum on servers
```

Comment:
This seems off: this says the spec requires a minimum but we use 9 as the maximum.

This needs revision.

Reply:
"client cannot send more than 9 since spec mandates 9 as minimum on the server side"